### DISCUSSION TITLE
Trying to understand the `typeof RedisConnection` signature on Queue constructor

### DISCUSSION THREAD

**Discussion by prescience-data** [taskforcesh/bullmq#1108] [Q&A] [answered]
I'm trying to create a shared redis connection for a small `Queue`, `QueueScheduler`, and `QueueEvents` group.

The docs which appear to be out of date seem to instruct creation of a new `IORedis` instance and then passing that in as `connection` in the options. 
https://docs.bullmq.io/guide/connections

This causes a `BullMQ: DEPRECATION WARNING!` which I assume is due to the `QueueBaseOptions` signature no longer accepting an instantiated redis connection:
https://github.com/taskforcesh/bullmq/blob/master/docs/gitbook/api/bullmq.queuebaseoptions.md

When I look through the code it seems like the new approach is to (maybe?) pass in a `RedisConnection` instance created like:
```typescript
import { RedisConnection, Queue } from "bullmq"
import { redisOptions, queueOptions, queueName } from "./config"

const connection = new RedisConnection(redisOptions, true)
const queue = new Queue(queueName, queueOptions, connection)
```

However the constructor signature for `Queue` seems to demand a `typeof RedisConnection` rather than `RedisConnection` which causes a `TS2345` type error as a `RedisConnection` instance `!==` the class constructor of `RedisInstance`.

https://github.com/taskforcesh/bullmq/blob/master/docs/gitbook/api/bullmq.queue._constructor_.md 

```
TS2345: Argument of type 'RedisConnection' is not assignable to parameter of type 'typeof RedisConnection'.   Type 'RedisConnection' is missing the following properties from type 'typeof RedisConnection': prototype, minimumVersion, waitUntilReady, getEventListeners, and 5 more.
```

This leads me to think that there is either two possibilities:

1) I am meant to extend the `RedisConnection` class somehow and provide the instruction to share the connection some other way, or;

2) The signature for `Queue._constructor_` is actually meant to be 
```typescript
constructor(name: string, opts?: QueueOptions, Connection?: RedisConnection); 
```
rather than the current `typeof RedisConnection`:
```typescript
constructor(name: string, opts?: QueueOptions, Connection?: typeof RedisConnection);
```

If the answer is 1) then any tips to get back on the right path would be awesome!

**Comment by manast** [comment, marked as answer]
The documentation is correct, you do not need to use the third argument, that is only used for implementing extensions that need different features than the ones provided by the standard RedisConnection class. You just need to pass an existing redis client as connection and then the class will reuse it as much as it can:
```ts
const connection = new Redis();
const queue = new Queue('myqueue', { connection });
```
